verification.py

```diff
@@ -5504,6 +5505,348 @@ def test_simulate_from_demography(self):
             pyplot.close("all")
 
 
+class SimulateAboveRoot(Test):
+    models = {
+        "SmcKApprox": msprime.SmcKApproxCoalescent(),
+        "Hudson": msprime.StandardCoalescent(),
+        "BetanCoalescent": msprime.BetaCoalescent(alpha=1.001),
+        "DiracCoalescent": msprime.DiracCoalescent(psi=0.9, c=10),
+    }
+    colors = list(mcolors.TABLEAU_COLORS.values())
+
+    def run_above_root_stats(self, initial_condition=False, **kwargs):
+        df_list = []
+
+        model = kwargs.get("model")
+        for stop_at_local_mrca in [True, False]:
+            kwargs["stop_at_local_mrca"] = stop_at_local_mrca
+            if initial_condition:
+                kwargs["model"] = [msprime.StandardCoalescent(duration=50), model]
+            else:
+                kwargs["model"] = model
+
+            logging.debug(f"Running: {kwargs}")
+            data = collections.defaultdict(list)
+            replicates = msprime.sim_ancestry(**kwargs)
+            for ts in replicates:
+                ts = (
+                    ts.simplify()
+                )  # we also simplify when the flag is True for consistency
+                t_mrca = np.zeros(ts.num_trees)
+                t_intervals = []
+                for tree in ts.trees():
+                    t_mrca[tree.index] = tree.time(tree.root)
+                    t_intervals.append(tree.interval)
+                data["tmrca_mean"].append(np.mean(t_mrca))
+                data["num_trees"].append(ts.num_trees)
+                data["intervals"].append(t_intervals)
+                data["model"].append(model)
+                data["stop_at_local_mrca"].append(stop_at_local_mrca)
+            df_list.append(pd.DataFrame(data))
+        return pd.concat(df_list)
+
+    def plot_above_root_stats(self, df):
+        df_above_root = df[df.stop_at_local_mrca == False]
+        df_stop_at_local_mrca = df[df.stop_at_local_mrca == True]
+        for stat in ["tmrca_mean", "num_trees"]:
+            plot_qq(df_above_root[stat], df_stop_at_local_mrca[stat])
+            f = self.output_dir / f"{stat}.png"
+            pyplot.savefig(f, dpi=72)
+            pyplot.close("all")
+
+        above_root_breakpoints = all_breakpoints_in_replicates(
```

Comment:
Don't bother with the breakpoints, any problems should come through in the number of trees.